### ISSUE TITLE
 RiotX isn't sending width / height data for uploaded image thumbnails.

### ISSUE THREAD

**Issue by starcraft66** [vector-im/element-android#1628]
RiotX isn't sending width / height data for uploaded image thumbnails, the event just has "w": 0, "h": 0.

This causes the image not to appear in Riot-iOS for example.

**Comment by SPK44** [comment]
I did some additional testing on this, and came up with some reproduction steps.

1. Click "+" to attach a file
2. Select "File", rather than "Gallery"
3. Select an image file
4. Choose "Send Image with Original Size"
5. Send the image

This was the only way I could reproduce the issue. I'll take a crack at debugging and fixing the issue soon.

**Comment by starcraft66** [comment]
Yeah, this is the way I always send pictures.

**Comment by SPK44** [comment]
For now, using the Gallery option, or unchecking the original size option will mitigate the issue.

**Comment by bmarty** [comment]
We (or at least I) had in mind that if you select "File", rather than "Gallery", and then you actually select an image it will send the image as a file, without previewing, without thumbnail, without resizing, and then without rendering of the image in the timeline (because the msg type would be `m.file`). But this is not what happenned anyway...

**Comment by SPK44** [comment]
I guess I should have consulted the dev team then to see if you wanted to dynamically select the data type to correct the user's action, or presume the users selection of file to be correct and send it without metadata.

 I noticed that the AttachmentMapper class would previously convert the File attachment to an image, audio, or video, if the Mime type suggested it. So I just moved the that Mime type correction higher up to the MultiPicker.

So another option would be to send images selected during file selection as if they were unpreviewable files. However, the original issue opener suggested that they always use the File button to send images. I imagine many other people (including me) do that to use a custom file browser application, rather than a gallery application.

TL;DR do the user's actually want to unpreviewable media objects in the timeline when they select "File" or do they just want to use thw file browser instead of the gallery.

**Comment by SPK44** [comment]
It looks like [this line change during a version upgrade](https://github.com/vector-im/element-android/commit/ff0f42900d4714afef65c226c218067dcafbac5b#diff-40c47fd843620c38e4415e5731d825aaacf4d74d6e400cc9b573409fa550392dR894) fixed this issue. And the latest version on the play store no longer is able to produce this bug.

Although I am not sure that this change was intentional since [this onFileResult callback was created in the same commit](https://github.com/vector-im/element-android/commit/ff0f42900d4714afef65c226c218067dcafbac5b#diff-bd4ee5082cfdf91e3fd82d51521a96a0ff43f37c5cf487ce73a7c505913adaffR110), and is currently unused.

I am going to assume that it was an intentional change though, and will close my pull request since this issue is no longer present.

**Comment by bmarty** [comment]
Cool, thanks for the follow up. I close this issue then.
